### ISSUE TITLE
Inspection suggest usage of static:: when the method name begins with static

### ISSUE THREAD

**Issue by thexpand** [kalessil/phpinspectionsea#626]
When a method is defined in a class via annotations like this: `@method StaticBlockManager staticBlock()` and later used withing the `$this` context `$this->staticBlock()->get("CONTACTS")` the inspections suggest that I use `static::` instead of `$this->`.

Here's an attachment:
![screenshot from 2017-11-07 10-12-42](https://user-images.githubusercontent.com/14979167/32483613-795cd746-c3a5-11e7-85a4-948d3e0f3f9e.png)

**Comment by kalessil** [comment]
Thank you for reporting @thexpand.
What happens if you apply the suggested change?

**Comment by thexpand** [comment]
@kalessil It becomes `static::staticBlock()->getTitle("CONTACTS")`

**Comment by kalessil** [comment]
And then, does application gets broken or keeps running?

**Comment by thexpand** [comment]
No, it's broken, because it is not a static method. I'm using ZendFramework and in the template, I'm using a fake class, where I write my `@method` doc blocks (for auto-suggest in the template). The reason why I'm doing this is because the view helper in ZendFramework are called via the magic method `__call()`.

**Comment by kalessil** [comment]
Oh, I see now. This is a bug in IDE itself:
- https://bitbucket.org/kalessil/phpinspectionsea/issues/300/false-positive-in-static-methods
- https://youtrack.jetbrains.com/issue/WI-33569

Please vote for it and I'll write a workaround for it.

**Comment by thexpand** [comment]
@kalessil I voted, but it seems that they have abandoned the issue. It's more than an year old.

**Comment by kalessil** [comment]
Ths. Once I have done with the workaround, I'll ask guys to close the issue.
